java-runtime/ftl-runtime/deployment/src/main/java/xyz/block/ftl/deployment/FTLCodeGenerator.java

```diff
@@ -274,52 +274,53 @@ private String toJavaName(String name) {
     }
 
     private TypeName toAnnotatedJavaTypeName(Type type, Map<Key, Type> typeAliasMap) {
-        var results = toJavaTypeName(type, typeAliasMap);
+        var results = toJavaTypeName(type, typeAliasMap, false);
         if (type.hasRef() || type.hasArray() || type.hasBytes() || type.hasString() || type.hasMap() || type.hasTime()) {
             return results.annotated(AnnotationSpec.builder(NotNull.class).build());
         }
         return results;
     }
 
-    private TypeName toJavaTypeName(Type type, Map<Key, Type> typeAliasMap) {
+    private TypeName toJavaTypeName(Type type, Map<Key, Type> typeAliasMap, boolean boxPrimitives) {
         if (type.hasArray()) {
             return ParameterizedTypeName.get(ClassName.get(List.class),
-                    toJavaTypeName(type.getArray().getElement(), typeAliasMap));
+                    toJavaTypeName(type.getArray().getElement(), typeAliasMap, false));
         } else if (type.hasString()) {
             return ClassName.get(String.class);
         } else if (type.hasOptional()) {
-            return toJavaTypeName(type.getOptional().getType(), typeAliasMap);
+            // Always box for optional, as normal primities can't be null
+            return toJavaTypeName(type.getOptional().getType(), typeAliasMap, true);
```

Comment:
if I'm not mistaken it appears that we are unwrapping the optional type and boxing it. What do you think about making use of `java.util.Optional[T]` instead in this instance?

Reply:
A boxed type maps exactly to the underlying FTL type (true, false, undefined). Optional adds an extra state (true, false, empty, the Optional field is null). 

Optional has never really hugely caught on, and it has a lot of drawbacks (Intellij actually warns against using it for parameter or field types). It was mostly used to indicate that a method may return a null value. In practice it solves very little, and you end up with an extra possible state that the Optional itself is null. It also does not play particularly well with both Kotlin and JSON mapping libraries, both of which are very relevant here. 

To make optional types work the way you would expect in Kotlin you need to use the @NotNull annotation, rather than Optional, which is why I am going for this approach.

Reply:
Will this get represented in Kotlin as a native optional?

Reply:
It should be yes, although medium term I might want to do an actual native kotlin code generator.

There is still more work around optional, I have filed https://github.com/TBD54566975/ftl/issues/2356, but in general kotlin uses the @NotNull annotation to tell the difference between Foo and Foo?.